lint checks fail:
```
[14:28:28] 
/home/travis/build/webrtc/samples/src/content/peerconnection/perfect-negotiation/js/main.js
  14:40  error  Strings must use singlequote            quotes
  15:51  error  Strings must use singlequote            quotes
  17:32  error  Strings must use singlequote            quotes
  22:28  error  A space is required after ','           comma-spacing
  22:31  error  A space is required after ','           comma-spacing
  22:38  error  A space is required after ','           comma-spacing
  22:41  error  A space is required after ','           comma-spacing
  42:48  error  A space is required after ','           comma-spacing
  42:51  error  A space is required after ','           comma-spacing
  45:48  error  A space is required after ','           comma-spacing
  45:51  error  A space is required after ','           comma-spacing
  49:46  error  Strings must use singlequote            quotes
  50:48  error  Strings must use singlequote            quotes
  52:42  error  A space is required after ','           comma-spacing
  52:45  error  A space is required after ','           comma-spacing
  52:52  error  A space is required after ','           comma-spacing
  52:55  error  A space is required after ','           comma-spacing
  68:8   error  'peer' is not defined                   no-undef
  75:42  error  A space is required after ','           comma-spacing
  75:44  error  A space is required after ','           comma-spacing
  75:49  error  A space is required after ','           comma-spacing
  75:51  error  A space is required after ','           comma-spacing
  76:45  error  A space is required after ','           comma-spacing
  76:47  error  A space is required after ','           comma-spacing
  76:52  error  A space is required after ','           comma-spacing
  76:54  error  A space is required after ','           comma-spacing
  80:16  error  'swapOnBoth' is defined but never used  no-unused-vars
  82:37  error  Strings must use singlequote            quotes
  83:39  error  Strings must use singlequote            quotes
  85:39  error  Strings must use singlequote            quotes
  86:37  error  Strings must use singlequote            quotes
/home/travis/build/webrtc/samples/src/content/peerconnection/perfect-negotiation/js/peer.js
   11:1   error  Missing JSDoc @return for function                                                                             valid-jsdoc
   11:1   error  Missing JSDoc for parameter 'other'                                                                            valid-jsdoc
   11:1   error  Missing JSDoc for parameter 'polite'                                                                           valid-jsdoc
   11:1   error  Missing JSDoc for parameter 'fail'                                                                             valid-jsdoc
   14:10  error  'peer' is defined but never used                                                                               no-unused-vars
   15:79  error  Strings must use singlequote                                                                                   quotes
   16:3   error  Expected { after 'if' condition                                                                                curly
   17:37  error  There should be no space after '{'                                                                             object-curly-spacing
   17:72  error  There should be no space before '}'                                                                            object-curly-spacing
   22:47  error  Strings must use singlequote                                                                                   quotes
   23:47  error  Strings must use singlequote                                                                                   quotes
   24:47  error  Strings must use singlequote                                                                                   quotes
   26:22  error  Unexpected space(s) after '{'                                                                                  block-spacing
   26:22  error  Statement inside of curly braces should be on next line                                                        brace-style
   26:46  error  Strings must use singlequote                                                                                   quotes
   26:60  error  Strings must use singlequote                                                                                   quotes
   26:83  error  Unexpected space(s) before '}'                                                                                 block-spacing
   26:83  error  Closing curly brace should be on the same line as opening curly brace or on the line after the previous block  brace-style
   29:11  error  Strings must use singlequote                                                                                   quotes
   36:1   error  Expected indentation of 12 spaces but found 10                                                                 indent
   36:53  error  There should be no space after '{'                                                                             object-curly-spacing
   36:86  error  Strings must use singlequote                                                                                   quotes
   36:97  error  There should be no space before '}'                                                                            object-curly-spacing
   38:1   error  Expected indentation of 12 spaces but found 10                                                                 indent
   38:29  error  Strings must use singlequote                                                                                   quotes
   38:38  error  There should be no space after '{'                                                                             object-curly-spacing
   38:71  error  Strings must use singlequote                                                                                   quotes
   38:82  error  There should be no space before '}'                                                                            object-curly-spacing
   42:52  error  Strings must use singlequote                                                                                   quotes
   43:49  error  Strings must use singlequote                                                                                   quotes
   45:49  error  Strings must use singlequote                                                                                   quotes
   48:49  error  Strings must use singlequote                                                                                   quotes
   50:49  error  Strings must use singlequote                                                                                   quotes
   58:11  error  Strings must use singlequote                                                                                   quotes
   62:25  error  Strings must use singlequote                                                                                   quotes
   62:42  error  There should be no space after '{'                                                                             object-curly-spacing
   62:54  error  There should be no space before '}'                                                                            object-curly-spacing
   62:72  error  There should be no space after '{'                                                                             object-curly-spacing
   62:84  error  There should be no space before '}'                                                                            object-curly-spacing
   63:5   error  Split 'let' declarations into multiple statements                                                              one-var
   65:25  error  Strings must use singlequote                                                                                   quotes
   67:13  error  Strings must use singlequote                                                                                   quotes
   68:9   error  'assert_equals' is not defined                                                                                 no-undef
   68:42  error  Strings must use singlequote                                                                                   quotes
   68:52  error  Strings must use singlequote                                                                                   quotes
   69:9   error  'assert_equals' is not defined                                                                                 no-undef
   69:43  error  Strings must use singlequote                                                                                   quotes
   72:9   error  'assert_equals' is not defined                                                                                 no-undef
   72:42  error  Strings must use singlequote                                                                                   quotes
   72:62  error  Strings must use singlequote                                                                                   quotes
   73:9   error  'assert_equals' is not defined                                                                                 no-undef
   73:49  error  Strings must use singlequote                                                                                   quotes
   73:58  error  Strings must use singlequote                                                                                   quotes
   74:21  error  There should be no space after '{'                                                                             object-curly-spacing
   74:56  error  There should be no space before '}'                                                                            object-curly-spacing
   81:31  error  There should be no space after '{'                                                                             object-curly-spacing
   81:39  error  There should be no space after '{'                                                                             object-curly-spacing
   81:69  error  There should be no space before '}'                                                                            object-curly-spacing
   81:71  error  There should be no space before '}'                                                                            object-curly-spacing
   88:15  error  'isStable' is never reassigned. Use 'const' instead                                                            prefer-const
   88:47  error  Strings must use singlequote                                                                                   quotes
   89:35  error  Strings must use singlequote                                                                                   quotes
   91:33  error  Strings must use singlequote                                                                                   quotes
   93:17  error  Strings must use singlequote                                                                                   quotes
   96:35  error  Strings must use singlequote                                                                                   quotes
   99:15  error  Strings must use singlequote                                                                                   quotes
   99:43  error  Strings must use singlequote                                                                                   quotes
  102:35  error  Strings must use singlequote                                                                                   quotes
  103:13  error  'assert_equals' is not defined                                                                                 no-undef
  103:46  error  Strings must use singlequote                                                                                   quotes
  103:67  error  Strings must use singlequote                                                                                   quotes
  104:13  error  'assert_equals' is not defined                                                                                 no-undef
  104:54  error  Strings must use singlequote                                                                                   quotes
  104:63  error  Strings must use singlequote                                                                                   quotes
  105:17  error  Strings must use singlequote                                                                                   quotes
  107:13  error  'assert_equals' is not defined                                                                                 no-undef
  107:46  error  Strings must use singlequote                                                                                   quotes
  107:56  error  Strings must use singlequote                                                                                   quotes
  108:13  error  'assert_equals' is not defined                                                                                 no-undef
  108:53  error  Strings must use singlequote                                                                                   quotes
  108:63  error  Strings must use singlequote                                                                                   quotes
  109:25  error  There should be no space after '{'                                                                             object-curly-spacing
  109:60  error  There should be no space before '}'                                                                            object-curly-spacing
  111:13  error  'assert_equals' is not defined                                                                                 no-undef
  111:54  error  Strings must use singlequote                                                                                   quotes
  111:64  error  Strings must use singlequote                                                                                   quotes
  112:13  error  'assert_equals' is not defined                                                                                 no-undef
  112:46  error  Strings must use singlequote                                                                                   quotes
  112:56  error  Strings must use singlequote                                                                                   quotes
  113:40  error  Strings must use singlequote                                                                                   quotes
  119:13  error  Expected { after 'if' condition                                                                                curly
  123:31  error  There should be no space after '{'                                                                             object-curly-spacing
  123:74  error  There should be no space before '}'                                                                            object-curly-spacing
```

Could these be addressed? Mostly style guide complaints

--- a/index.html
+++ b/index.html
@@ -132,6 +132,8 @@
         <ul>
             <li><a href="src/content/peerconnection/pc1/">Basic peer connection demo</a></li>
 
+            <li><a href="src/content/peerconnection/perfect-negotiation/">Peer connection using Perfect Negotiation</a></li>
+
             <li><a href="src/content/peerconnection/audio/">Audio-only peer connection demo</a></li>
 
             <li><a href="src/content/peerconnection/bandwidth/">Change bandwidth on the fly</a></li>

--- a/src/content/peerconnection/perfect-negotiation/css/main.css
+++ b/src/content/peerconnection/perfect-negotiation/css/main.css
@@ -1,0 +1,48 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+button {
+  margin: 0 20px 0 0;
+  width: 83px;
+}
+
+button#hangupButton {
+    margin: 0;
+}
+
+video {
+  --width: 45%;
+  width: var(--width);
+  height: calc(var(--width) * 0.75);
+  margin: 0 0 20px 0;
+  vertical-align: top;
+}
+
+video#localVideo {
+  margin: 0 20px 20px 0;
+}
+
+div.box {
+  margin: 1em;
+}
+
+@media screen and (max-width: 400px) {
+  button {
+    width: 83px;
+    margin: 0 11px 10px 0;
+  }
+
+  video {
+    height: 90px;
+    margin: 0 0 10px 0;
+    width: calc(50% - 7px);
+  }
+  video#localVideo {
+    margin: 0 10px 20px 0;
+  }
+
+}

--- a/src/content/peerconnection/perfect-negotiation/index.html
+++ b/src/content/peerconnection/perfect-negotiation/index.html
@@ -1,0 +1,81 @@
+<!DOCTYPE html>
+<!--
+ *  Copyright (c) 2020 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+-->
+<html>
+<head>
+
+    <meta charset="utf-8">
+    <meta name="description" content="WebRTC Javascript code samples">
+    <meta name="viewport" content="width=device-width, user-scalable=yes, initial-scale=1, maximum-scale=1">
+    <meta itemprop="description" content="Client-side WebRTC code samples">
+    <meta itemprop="image" content="../../../images/webrtc-icon-192x192.png">
+    <meta itemprop="name" content="WebRTC code samples">
+    <meta name="mobile-web-app-capable" content="yes">
+    <meta id="theme-color" name="theme-color" content="#ffffff">
+
+    <base target="_blank">
+
+    <title>Perfect Negotiation</title>
+
+    <link rel="icon" sizes="192x192" href="../../../images/webrtc-icon-192x192.png">
+    <link href="//fonts.googleapis.com/css?family=Roboto:300,400,500,700" rel="stylesheet" type="text/css">
+    <link rel="stylesheet" href="../../../css/main.css"/>
+    <link rel="stylesheet" href="css/main.css"/>
+
+</head>
+
+<body>
+
+<div id="container">
+    <h1><a href="//webrtc.github.io/samples/" title="WebRTC samples homepage">WebRTC samples</a>
+        <span>Perfect Negotiation</span></h1>
+
+    <p>This sample shows how to setup a connection between two peers using
+        <a href="https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection">RTCPeerConnection</a>
+        with the <a href="https://w3c.github.io/webrtc-pc/#perfect-negotiation-example">Perfect Negotiation</a>
+        usage pattern.</p>
+
+    <p>Perfect Negotiation supports both endpoints sending offers. The pattern intelligently handles the
+        situation of "glare" (both peers making an offer at the same time, causing a collision) by having
+        one peer be "polite" and the other peer be "impolite". In the event of an offer collision, the
+        polite peer rolls back its offer in order to process the impolite peer's incoming offer. Once
+        back to the stable signaling state, the polite peer's <code>onnegotiationneeded</code> fires again and a
+        follow-up O/A is completed.</p>
+
+    <p>Click both peers' Start button to create local streams.
+        Then press the Swap Sending Track button to modify which transceiver is sending;
+        this will be negotiated and displayed as a remote track on the other peer's iframe.</p>
+
+    <div id="iframes">
+        <iframe id="polite"></iframe>
+        <iframe id="impolite"></iframe>
+    </div>
+
+    <div id="buttons">
+        <button onclick="swapOnBoth(true)">Swap on both (glare): Polite to offer first</button>
+        <button onclick="swapOnBoth(false)">Swap on both (glare): Impolite to offer first</button>
+    </div>
+
+    <p>The JavaScript console shows logs for the negotiation steps.</p>
+
+    <p>For more information about RTCPeerConnection, see <a href="http://www.html5rocks.com/en/tutorials/webrtc/basics/"
+                                                            title="HTML5 Rocks article about WebRTC by Sam Dutton">Getting
+        Started With WebRTC</a>.</p>
+
+
+    <a href="https://github.com/webrtc/samples/tree/gh-pages/src/content/peerconnection/perfect-negotiation"
+       title="View source for this page on GitHub" id="viewSource">View source on GitHub</a>
+
+</div>
+
+<script src="https://webrtc.github.io/adapter/adapter-latest.js"></script>
+<script src="js/main.js" type="module" async></script>
+
+<script src="../../../js/lib/ga.js"></script>
+</body>
+</html>

--- a/src/content/peerconnection/perfect-negotiation/js/main.js
+++ b/src/content/peerconnection/perfect-negotiation/js/main.js
@@ -1,0 +1,118 @@
+/*
+ *  Copyright (c) 2020 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+import {peer} from './peer.js';
+
+const politeIframe = document.getElementById('polite');
+const impoliteIframe = document.getElementById('impolite');
+let counter = 0;
+
+/**
+ * @param {Window} target
+ * @param {string} cmd
+ */
+export async function run(target, cmd) {
+  const id = `result${counter++}`;
+  target.postMessage({run: {cmd, id}}, '*');
+  return new Promise(resolve => void window.addEventListener('message', function listen({data}) {
+    if (!(id in data)) return;
+    window.removeEventListener('message', listen);
+    resolve(data[id]);
+  }));
+}
+
+/**
+ * @param {number} r1 Video 1 Red channel level [0-1]
+ * @param {number} g1 Video 1 Green channel level [0-1]
+ * @param {number} b1 Video 1 Blue channel level [0-1]
+ * @param {number} r2 Video 2 Red channel level [0-1]
+ * @param {number} g2 Video 2 Green channel level [0-1]
+ * @param {number} b2 Video 2 Blue channel level [0-1]
+ */
+export function startLocalVideo(r1, g1, b1, r2, g2, b2) {
+  const whiteNoise = (width, height, r, g, b) => {
+    const canvas = Object.assign(document.createElement('canvas'), {width, height});
+    const ctx = canvas.getContext('2d');
+    ctx.fillRect(0, 0, width, height);
+    const p = ctx.getImageData(0, 0, width, height);
+    const draw = () => {
+      for (let i = 0; i < p.data.length; i++) {
+        const color = Math.random() * 255;
+        p.data[i++] = color * r;
+        p.data[i++] = color * g;
+        p.data[i++] = color * b;
+      }
+      ctx.putImageData(p, 0, 0);
+      requestAnimationFrame(draw);
+    };
+    requestAnimationFrame(draw);
+    return canvas.captureStream();
+  };
+  const localVideo1 = document.getElementById('localVideo1');
+  localVideo1.srcObject = whiteNoise(32, 32, r1, g1, b1);
+  localVideo1.play();
+  const localVideo2 = document.getElementById('localVideo2');
+  localVideo2.srcObject = whiteNoise(32, 32, r2, g2, b2);
+  localVideo2.play();
+}
+
+/**
+ * @param {HTMLIFrameElement} el
+ * @param {boolean} polite
+ * @param {number} r1 Video 1 Red channel level [0-1]
+ * @param {number} g1 Video 1 Green channel level [0-1]
+ * @param {number} b1 Video 1 Blue channel level [0-1]
+ * @param {number} r2 Video 2 Red channel level [0-1]
+ * @param {number} g2 Video 2 Green channel level [0-1]
+ * @param {number} b2 Video 2 Blue channel level [0-1]
+ */
+async function setupIframe(el, polite, r1, g1, b1, r2, g2, b2) {
+  el.srcdoc = `<!DOCTYPE html>
+      <html>
+      <body>
+      <h3 style="font-family: 'Roboto', sans-serif; font-weight: 400;">${polite ? 'Polite' : 'Impolite'} Peer's iframe</h3>
+      <div id="buttons">
+        <button onclick="(${startLocalVideo.toString()})(${r1},${g1},${b1},${r2},${g2},${b2});" id="${polite ? 'politeStart' : 'impoliteStart'}">Start</button>
+        <button onclick="window.parent.run(window, 'swapTransceivers');">Swap Sending Track</button>
+      </div>
+      <p id="videos">
+        <video id="localVideo1" autoplay></video>
+        <video id="localVideo2" autoplay></video>
+        <video id="remoteVideo" autoplay></video>
+      </p>
+      </body>
+      <script>
+        (${peer.toString()})(window.parent.document.getElementById("${polite ? 'impolite' : 'polite'}").contentWindow, ${polite});
+      </script>
+      <html>`;
+  await new Promise(resolve => el.onload = resolve);
+}
+
+export async function swapOnBoth(politeFirst) { // eslint-disable-line no-unused-vars
+  if (politeFirst) {
+    run(politeIframe.contentWindow, 'swapTransceivers');
+    run(impoliteIframe.contentWindow, 'swapTransceivers');
+  } else {
+    run(impoliteIframe.contentWindow, 'swapTransceivers');
+    run(politeIframe.contentWindow, 'swapTransceivers');
+  }
+}
+
+async function setupIframes() {
+  await setupIframe(politeIframe, true, 0, 1, 0, 0, 1, 1);
+  await setupIframe(impoliteIframe, false, 1, 0, 0, 1, 0, 1);
+}
+
+window.run = run;
+window.swapOnBoth = swapOnBoth;
+window.peer = peer;
+export {peer};
+
+setupIframes();

--- a/src/content/peerconnection/perfect-negotiation/js/peer.js
+++ b/src/content/peerconnection/perfect-negotiation/js/peer.js
@@ -1,0 +1,136 @@
+/*
+ *  Copyright (c) 2020 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+/**
+ * Establishes a Peer Connection with `other` using the Perfect Negotiation pattern.
+ * @param {Window} other
+ * @param {boolean} polite
+ * @param {function(Error): void} fail
+ * @return {RTCPeerConnection} Peer Connection
+ */
+function peer(other, polite, fail = undefined) { // eslint-disable-line no-unused-vars
+  if (!fail) fail = e => void send(window.parent, {error: `${e.name}: ${e.message}`});
+  const send = (target, msg) => void target.postMessage(JSON.parse(JSON.stringify(msg)), '*');
+  const log = str => void console.log(`[${polite ? 'POLITE' : 'IMPOLITE'}] ${str}`);
+  const assert_equals = !window.assert_equals ?
+      (a, b, msg) => a === b || void fail(new Error(`${msg} expected ${b} but got ${a}`)) :
+      window.assert_equals;
+  const pc = new RTCPeerConnection();
+  const localVideo1 = document.getElementById('localVideo1');
+  const localVideo2 = document.getElementById('localVideo2');
+  const remoteVideo = document.getElementById('remoteVideo');
+  const transceiversForSending = [];
+  const commands = {
+    async swapTransceivers() {
+      log('swapTransceivers');
+      const stream1 = localVideo1.srcObject;
+      const stream2 = localVideo2.srcObject;
+      if (transceiversForSending.length == 0) {
+        // This is the first time swapTransceivers is called.
+        // Add the initial transceivers, which are remembered for future swaps.
+        transceiversForSending.push(
+            pc.addTransceiver(stream1.getTracks()[0], {streams: [stream1], direction: 'sendonly'}));
+        transceiversForSending.push(
+            pc.addTransceiver('video', {streams: [stream2], direction: 'inactive'}));
+        return;
+      }
+      // We have sent before. Swap which transceiver is the sending one.
+      if (transceiversForSending[0].direction == 'sendonly') {
+        transceiversForSending[0].direction = 'inactive';
+        transceiversForSending[0].sender.replaceTrack(null);
+        transceiversForSending[1].direction = 'sendonly';
+        transceiversForSending[1].sender.replaceTrack(stream2.getTracks()[0]);
+      } else {
+        transceiversForSending[1].direction = 'inactive';
+        transceiversForSending[1].sender.replaceTrack(null);
+        transceiversForSending[0].direction = 'sendonly';
+        transceiversForSending[0].sender.replaceTrack(stream1.getTracks()[0]);
+      }
+    },
+  };
+  try {
+    pc.ontrack = e => {
+      log('ontrack');
+      remoteVideo.srcObject = new MediaStream();
+      remoteVideo.srcObject.addTrack(e.track);
+    };
+    pc.onicecandidate = ({candidate}) => void send(other, {candidate});
+    let makingOffer = false;
+    let ignoreOffer = false;
+    let srdAnswerPending = false;
+    pc.onnegotiationneeded = async () => {
+      try {
+        log('SLD due to negotiationneeded');
+        assert_equals(pc.signalingState, 'stable', 'negotiationneeded always fires in stable state');
+        assert_equals(makingOffer, false, 'negotiationneeded not already in progress');
+        makingOffer = true;
+        await pc.setLocalDescription();
+        assert_equals(pc.signalingState, 'have-local-offer', 'negotiationneeded not racing with onmessage');
+        assert_equals(pc.localDescription.type, 'offer', 'negotiationneeded SLD worked');
+        send(other, {description: pc.localDescription});
+      } catch (e) {
+        fail(e);
+      } finally {
+        makingOffer = false;
+      }
+    };
+    window.onmessage = async ({data: {description, candidate, run}}) => {
+      try {
+        if (description) {
+          // If we have a setRemoteDescription() answer operation pending, then
+          // we will be "stable" by the time the next setRemoteDescription() is
+          // executed, so we count this being stable when deciding whether to
+          // ignore the offer.
+          const isStable =
+              pc.signalingState == 'stable' ||
+              (pc.signalingState == 'have-local-offer' && srdAnswerPending);
+          ignoreOffer =
+              description.type == 'offer' && !polite && (makingOffer || !isStable);
+          if (ignoreOffer) {
+            log('glare - ignoring offer');
+            return;
+          }
+          srdAnswerPending = description.type == 'answer';
+          log(`SRD(${description.type})`);
+          await pc.setRemoteDescription(description);
+          srdAnswerPending = false;
+          if (description.type == 'offer') {
+            assert_equals(pc.signalingState, 'have-remote-offer', 'Remote offer');
+            assert_equals(pc.remoteDescription.type, 'offer', 'SRD worked');
+            log('SLD to get back to stable');
+            await pc.setLocalDescription();
+            assert_equals(pc.signalingState, 'stable', 'onmessage not racing with negotiationneeded');
+            assert_equals(pc.localDescription.type, 'answer', 'onmessage SLD worked');
+            send(other, {description: pc.localDescription});
+          } else {
+            assert_equals(pc.remoteDescription.type, 'answer', 'Answer was set');
+            assert_equals(pc.signalingState, 'stable', 'answered');
+            pc.dispatchEvent(new Event('negotiated'));
+          }
+        } else if (candidate) {
+          try {
+            await pc.addIceCandidate(candidate);
+          } catch (e) {
+            if (!ignoreOffer) throw e;
+          }
+        } else if (run) {
+          send(window.parent, {[run.id]: await commands[run.cmd]() || 0});
+        }
+      } catch (e) {
+        fail(e);
+      }
+    };
+  } catch (e) {
+    fail(e);
+  }
+  return pc;
+}
+
+export {peer};

--- a/src/content/peerconnection/perfect-negotiation/test/test.js
+++ b/src/content/peerconnection/perfect-negotiation/test/test.js
@@ -1,0 +1,36 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+/* eslint-env node */
+'use strict';
+export default {
+  'It should have select elements for each media device': (browser) => {
+    const path = '/src/content/peerconnection/perfect-negotiation/index.html';
+    const url = 'file://' + process.cwd() + path;
+
+    browser.url(url).waitForElementVisible('#startButton', 1000, 'Check that the startButton button is visible');
+    browser.waitForReadyState('#localVideo', 0, 1000);
+    browser.waitForReadyState('#remoteVideo', 0, 1000);
+    browser.expect.element('#callButton').to.not.be.enabled.before(1000);
+    browser.expect.element('#hangupButton').to.not.be.enabled.before(1000);
+    browser.click('#startButton');
+    browser.expect.element('#startButton').to.not.be.enabled.before(1000);
+    browser.expect.element('#callButton').to.be.enabled.before(1000);
+    browser.expect.element('#hangupButton').to.not.be.enabled.before(1000);
+    browser.waitForReadyState('#localVideo', 4, 1000);
+    browser.click('#callButton');
+    browser.waitForReadyState('#remoteVideo', 4, 1000);
+    browser.expect.element('#startButton').to.not.be.enabled.before(1000);
+    browser.expect.element('#callButton').to.not.be.enabled.before(1000);
+    browser.expect.element('#hangupButton').to.be.enabled.before(1000);
+    browser.click('#hangupButton');
+    browser.expect.element('#startButton').to.not.be.enabled.before(1000);
+    browser.expect.element('#callButton').to.be.enabled.before(1000);
+    browser.expect.element('#hangupButton').to.not.be.enabled.before(1000);
+    browser.end();
+  }
+};

